net/tlsdial/tlsdial.go

```diff
@@ -99,6 +115,12 @@ func Config(host string, base *tls.Config) *tls.Config {
 	return conf
 }
 
+func certIsSelfSigned(cert *x509.Certificate) bool {
```

Comment:
oh, this is tighter than I imagined it'd be at first reading of the call site.

or does this also cover signatures with internal corp CAs?

Reply:
It doesn't, for now; as a follow-up I want to import all public root CAs and verify against that set so we can detect non-public CA certs.